src/libraries/Common/src/SourceGenerators/TypeRef.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
+
+namespace DotnetRuntime.SourceGenerators
+{
+    /// <summary>
+    /// An equatable value representing type identity.
+    /// </summary>
+    [DebuggerDisplay("Name={Name}")]
+    public sealed class TypeRef : IEquatable<TypeRef>
+    {
+        private static readonly SymbolDisplayFormat s_minimalDisplayFormat = new SymbolDisplayFormat(
+            globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+            typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypes,
+            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
+
+        public TypeRef(ITypeSymbol type)
+        {
+            (Name, Namespace, GlobalDisplayString, FullyQualifiedDisplayString, MinimalDisplayString) = GetIdentifierInfo(type);
+            IsValueType = type.IsValueType;
+            TypeKind = type.TypeKind;
+            SpecialType = type.SpecialType;
+        }
+
+        public string Name { get; }
+
+        public string? Namespace { get; }
+
+        /// <summary>
+        /// Fully qualified assembly name, prefixed with "global::", e.g. global::System.Numerics.BigInteger.
+        /// </summary>
+        public string GlobalDisplayString { get; }
```

Comment:
This property could be constructed on-demand by prefixing `global::` on the FQN property.